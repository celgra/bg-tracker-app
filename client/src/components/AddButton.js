import React from 'react';

import './AddButton.css';

const AddButton = (props) => {
    if(props.isAddingResult){
        return null;
    } else {
        return (
            <button
            type="button"
            className="btn btn-primary add-button"
            onClick={props.addResult}>
                Add
            </button>
        );
    }
};

export default AddButton;
