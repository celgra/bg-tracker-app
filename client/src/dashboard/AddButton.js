import React from 'react';

import './AddButton.css';

const AddButton = (props) => {
    const css = `btn btn-primary add-button ${props.css} ${props.isAddingResult ? 'disabled' : ''}`;

    return (
        <button
        type='button'
        className={css}
        onClick={props.addResult}>
            Add
        </button>
    );
};

export default AddButton;
