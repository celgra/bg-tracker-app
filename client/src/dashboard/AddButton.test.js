import React from 'react';
import AddButton from './AddButton';
import renderer from 'react-test-renderer';

test('Button', () => {
  const component = renderer.create(
    <AddButton css={'test'} isAddingResult={true}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});