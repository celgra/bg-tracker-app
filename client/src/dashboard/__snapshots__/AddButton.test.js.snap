// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<button
  className="btn btn-primary add-button test disabled"
  onClick={undefined}
  type="button"
>
  Add
</button>
`;
